pkg/build/api/v1beta3/types.go

```diff
@@ -382,6 +386,21 @@ type SourceBuildStrategy struct {
 	ForcePull bool `json:"forcePull,omitempty" description:"forces the source build to pull the image if true"`
 }
 
+// BuildPostCommitSpec holds a build post commit hook specification.
+type BuildPostCommitSpec struct {
+	// Shell is an action that executes a shell script in a container
+	// running the build output image.
+	Shell string `json:"shell,omitempty" description:"shell script to be executed in a container running the build output image"`
```

Comment:
I agree concatenation is not valuable in and of itself - but I'd like to hear counterarguments to the arguments handling issue if you have them.  I think `shellArgs: ["bundle", "exec, "rake", "test"]` is at not quite as obvious as `shell: "bundle exec rake test"`, but `shellArgs: ["mvn", "-Dapp=$(POD_NAME)"]` is clearer than `shell: "mvn -Dapp=$(POD_NAME)"` which could easily have spaces and be wrong.